test/smartwallet/smartWallet.test.ts

```diff
@@ -1,943 +1,519 @@
-//             from: fundedAccount,
-//           })
-//         );
-//       });
-
-//       it('Should revert call to directExecute with no funded account as parameter', async () => {
-//         await assert.isRejected(
-//           smartWallet.directExecute(recipient.address, recipientFunction),
-//           'Not the owner of the SmartWallet',
-//           'Error while validating the data'
-//         );
-//       });
-
-//       it('Should revert call to directExecute with INVALID_ARGUMENT as parameter', async () => {
-//         await assert.isRejected(
-//           smartWallet.directExecute('', recipientFunction, {
-//             from: fundedAccount,
-//           }),
-//           'INVALID_ARGUMENT',
-//           'Error while validating the data'
-//         );
-//       });
-
-//       it('Should call successfully to method directExecute through node funded account', async () => {
-//         assert.isFalse(await smartWallet.isInitialized());
-
-//         await smartWallet.initialize(
-//           fundedAccount,
-//           token.address,
-//           worker,
-//           '0',
-//           '400000'
-//         );
-
-//         assert.isTrue(await smartWallet.isInitialized());
-//         const initialNonce = await smartWallet.nonce();
-
-//         await assert.isFulfilled(
-//           smartWallet.directExecute(recipient.address, recipientFunction, {
-//             from: fundedAccount,
-//           })
-//         );
-//         assert.equal(
-//           (await smartWallet.nonce()).toString(),
-//           initialNonce.toString(),
-//           'Call to direct execute should NOT increment nonce'
-//         );
-//       });
-//     });
-//   }
-// );
-
-// /* TODO - ADD tests for recover method */
+import { ethers as hardhat } from 'hardhat';
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import chai, { expect} from 'chai';
+import { FakeContract, smock } from '@defi-wonderland/smock';
+import chaiAsPromised from 'chai-as-promised';
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { TypedDataUtils } from '@metamask/eth-sig-util';
+import {
+    getLocalEip712Signature,
+    TypedRequestData} from '../utils/EIP712Utils';
+import { SignTypedDataVersion } from '@metamask/eth-sig-util';
+import { Wallet } from 'ethers';
+import { SmartWallet } from 'typechain-types';
+import { BaseProvider } from '@ethersproject/providers';
+import { EnvelopingTypes, IForwarder } from 'typechain-types/contracts/RelayHub';
+
+chai.use(smock.matchers);
+chai.use(chaiAsPromised);
+
+const ZERO_ADDRESS = hardhat.constants.AddressZero;
+const ONE_FIELD_IN_BYTES = 32;
+
+type ForwardRequest = IForwarder.ForwardRequestStruct;
+type RelayData = EnvelopingTypes.RelayDataStruct;
+type RelayRequest = EnvelopingTypes.RelayRequestStruct;
+
+describe('SmartWallet', function(){
+    function createRequest(
+        request: Partial<ForwardRequest>,
+        relayData: Partial<RelayData>
```

Comment:
can we do the `relayData: Partial<RelayData>` optional?

Reply:
Done.